### PR TITLE
TST: Decoupled more xlrd reading tests from openpyxl

### LINKED ISSUE #27111: TST: openpyxl tests fail if xlrd is not installed
If you run the `tests/io/excel/test_readers.py` with only openpyxl installed, the xlrd tests are skipped, but some of the openpyxl tests still fail on needing xlrd.

eg:
```
____________________________________________________________ TestExcelFileRead.test_read_excel_engine_value[openpyxl-.xlsm-None] _____________________________________________________________

self = <pandas.tests.io.excel.test_readers.TestExcelFileRead object at 0x7fca3bb6d198>, read_ext = '.xlsm', excel_engine = None

    @pytest.mark.parametrize('excel_engine', [
        'xlrd',
        None
    ])
    def test_read_excel_engine_value(self, read_ext, excel_engine):
        # GH 26566
>       xl = ExcelFile("test1" + read_ext, engine=excel_engine)

../excel/test_readers.py:853: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../io/excel/_base.py:791: in __init__
    self._reader = self._engines[engine](self._io)
../../../io/excel/_xlrd.py:21: in __init__
    import_optional_dependency("xlrd", extra=err_msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

name = 'xlrd', extra = 'Install xlrd >= 1.0.0 for Excel support', raise_on_missing = True, on_version = 'raise'

    def import_optional_dependency(
        name: str,
        extra: str = "",
        raise_on_missing: bool = True,
        on_version: str = "raise",
    ):
        """
        Import an optional dependency.
    
        By default, if a dependency is missing an ImportError with a nice
        message will be raised. If a dependency is present, but too old,
        we raise.
    
        Parameters
        ----------
        name : str
            The module name. This should be top-level only, so that the
            version may be checked.
        extra : str
            Additional text to include in the ImportError message.
        raise_on_missing : bool, default True
            Whether to raise if the optional dependency is not found.
            When False and the module is not present, None is returned.
        on_version : str {'raise', 'warn'}
            What to do when a dependency's version is too old.
    
            * raise : Raise an ImportError
            * warn : Warn that the version is too old. Returns None
            * ignore: Return the module, even if the version is too old.
              It's expected that users validate the version locally when
              using ``on_version="ignore"`` (see. ``io/html.py``)
    
        Returns
        -------
        maybe_module : Optional[ModuleType]
            The imported module, when found and the version is correct.
            None is returned when the package is not found and `raise_on_missing`
            is False, or when the package's version is too old and `on_version`
            is ``'warn'``.
        """
        try:
            module = importlib.import_module(name)
        except ImportError:
            if raise_on_missing:
>               raise ImportError(message.format(name=name, extra=extra)) from None
E               ImportError: Missing optional dependency 'xlrd'. Install xlrd >= 1.0.0 for Excel support Use pip or conda to install xlrd.
```

The actual excel reading (interactively) is not affected, so certainly not a blocking issue.

### PR DESCRIPTION
**Pull request by WillAyd**
- [X] closes #27111
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

@simonjayhawkins

### CODE REVIEW

File: pandas/tests/io/excel/test_readers.py [hunk 1]
```diff
@@ -736,39 +735,44 @@ class TestExcelFileRead:
...
+            pytest.skip()
+
         func = partial(pd.ExcelFile, engine=request.param)
         monkeypatch.chdir(datapath("io", "data"))
         monkeypatch.setattr(pd, 'ExcelFile', func)
 
     def test_excel_passes_na(self, read_ext):
 
-        excel = ExcelFile('test4' + read_ext)
+        excel = pd.ExcelFile('test4' + read_ext)
```
Comment (WillAyd) [inline pandas/tests/io/excel/test_readers.py]: We were monkeypatching `pd.ExcelFile` but used `ExcelFile` imported directly in tests which was causing this to run exclusive of the specified engine. Figured easiest just to stick with the pd namespace

File: pandas/tests/io/excel/test_readers.py [hunk 2]
```diff
@@ -736,39 +735,44 @@ class TestExcelFileRead:
...
 
         parsed = pd.read_excel(excel, 'Sheet1', keep_default_na=False,
                                na_values=['apple'])
         expected = DataFrame([['NA'], [1], ['NA'], [np.nan], ['rabbit']],
                              columns=['Test'])
         tm.assert_frame_equal(parsed, expected)
 
+        excel = pd.ExcelFile('test4' + read_ext)
```
Comment (WillAyd) [inline pandas/tests/io/excel/test_readers.py]: openpyxl seems to exhaust a `pd.ExcelFile` after reading whereas xlrd does not. I don't think we really have a preference here and I can't imagine that the xlrd behavior is necessarily required so just created a new pd.ExcelFile after each use

File: pandas/tests/io/excel/test_readers.py [hunk 3]
```diff
@@ -787,70 +791,67 @@ def test_unexpected_kwargs_raises(self, read_ext, arg):
...
-        with pytest.raises(ValueError, match=msg):
-            pd.read_excel(xl, engine='openpyxl')
+@td.skip_if_no("openpyxl")
+@pytest.mark.parametrize('excel_engine', [
+    pytest.param('xlrd', marks=pytest.mark.skipif(
+        not td.safe_import("xlrd"), reason="no xlrd")),
+    pytest.param(None, marks=pytest.mark.skipif(
+        not td.safe_import("xlrd"), reason="no xlrd")),
+])
+def test_conflicting_excel_engines(read_ext, excel_engine, datapath):
```
Comment (WillAyd) [inline pandas/tests/io/excel/test_readers.py]: This didn't really follow the same rules for parametrization as the rest of the items in `TestExcelFileRead`so figured clearest to remove from class

File: pandas/tests/io/excel/test_readers.py [hunk 4]
```diff
@@ -778,7 +782,7 @@ def test_excel_passes_na(self, read_ext):
     @pytest.mark.parametrize('arg', ['sheet', 'sheetname', 'parse_cols'])
     def test_unexpected_kwargs_raises(self, read_ext, arg):
         # gh-17964
-        excel = ExcelFile('test1' + read_ext)
+        excel = pd.ExcelFile('test1' + read_ext)
```
Comment (jreback) [inline pandas/tests/io/excel/test_readers.py]: do we normally need to close the file? use in a context manager?

Comment (WillAyd) [reply pandas/tests/io/excel/test_readers.py]: It looks like it depends on what type of object ExcelFile is interacting with but ultimately converted usage to a context manager to be safe

File: pandas/tests/io/excel/test_writers.py [hunk 5]
```diff
@@ -250,6 +250,7 @@ class and any subclasses, on account of the `autouse=True`
         set_option(option_name, prev_engine)  # Roll back option change
 
 
+@td.skip_if_no('xlrd')
```
Comment (WillAyd) [inline pandas/tests/io/excel/test_writers.py]: The writer tests still assume xlrd to be there. Will fix up in a follow up to address proper parametrization here (have focused more on readers given recent PRs)

@simonjayhawkins

**Review by jorisvandenbossche — APPROVED** [review]
Tests pass now locally without having xlrd installed!

**Comment by simonjayhawkins** [comment]
This PR results in an additional 2 warnings. although 8 of the same were introduced in #25092.

```
pandas/tests/io/excel/test_readers.py::TestReaders::test_reader_special_dtypes[openpyxl-.xlsx]
pandas/tests/io/excel/test_readers.py::TestReaders::test_reader_special_dtypes[openpyxl-.xlsm]
pandas/tests/io/excel/test_readers.py::TestReaders::test_reading_all_sheets[openpyxl-.xlsx]
pandas/tests/io/excel/test_readers.py::TestReaders::test_reading_all_sheets[openpyxl-.xlsm]
pandas/tests/io/excel/test_readers.py::TestReaders::test_reading_multiple_specific_sheets[openpyxl-.xlsx]
pandas/tests/io/excel/test_readers.py::TestReaders::test_reading_multiple_specific_sheets[openpyxl-.xlsm]
pandas/tests/io/excel/test_readers.py::TestReaders::test_read_excel_squeeze[openpyxl-.xlsx]
pandas/tests/io/excel/test_readers.py::TestReaders::test_read_excel_squeeze[openpyxl-.xlsm]
pandas/tests/io/excel/test_readers.py::TestExcelFileRead::test_excel_passes_na[openpyxl-.xlsx]
pandas/tests/io/excel/test_readers.py::TestExcelFileRead::test_excel_passes_na[openpyxl-.xlsm]
  C:\Users\simon\Anaconda3\envs\pandas-dev\lib\site-packages\openpyxl\worksheet\_reader.py:296: UserWarning: Unknown extension is not supported and will be removed
    warn(msg)
```

**Comment by jorisvandenbossche** [comment]
@simonjayhawkins that should be solved by https://github.com/pandas-dev/pandas/pull/27122

**Review by simonjayhawkins — APPROVED** [review]

**Pull request merged**